### ISSUE TITLE
Publication verification and more

### ISSUE THREAD

**Issue by JavierSegoviaCordoba** [gradle-nexus/publish-plugin#113]
First of all, I would like to add this issue to a discussion, but it is not available in this repo I think.

@LouisCAD has an interesting setup in Splitties repo which has two features I am missing:

- Publication verification (he created his own implementation).
- Use the same repository for multiple machines (he is using a few actions created by @martinbonnin).

## Publication verification

I haven't all the details, but it implements the artifacts in a dummy project from the staging stage, and if they are correct, then they will be released. Personally, due to Sonatype overloads, I am not confident with automatically releasing the artifacts, this verification would allow me to be confident and avoid visiting the Nexus console anymore.

## Share the same repository (KMP projects)

There is an open issue for this one (I think I was the author even) but as I said I would want to comment on this in a discussion topic (please open discussions! :D)

In Splitties you can see how @LouisCAD uses the @martinbonnin actions to publish KMP libraries sharing the same repository, so converting those actions to Gradle tasks in this plugin should be really great.

## Summary

This is a graphic about how it looks:

![image](https://user-images.githubusercontent.com/7463564/130614853-da231e53-dcb2-45cb-b833-9aa38ea69f9b.png)

Every step is self-explanatory except `finalize`, this step is checking if the publication is correct, if it is correct, the repo will be released, if not, it will be dropped.

**Comment by tynn** [comment]
I think this could be achieved by having the repository meta data persisted as an output of the initialization task. I'd like to see this for the same reasons as well. Do you already have plans for such a behavior?

**Comment by marcphilipp** [comment]
@tynn I wouldn't object having the initialization task write a metadata file, e.g. in JSON format. I'm not sure it would solve @JavierSegoviaCordoba's use case, though, since in his case verification happens on different machines/VMs, doesn't it?

**Comment by tynn** [comment]
@marcphilipp it depends. Writing the file itself would only solve it within the same run. But eventually you should be able to provide it from previous builds or even provide it as an input from outside (a property or file maybe).

I'll try to get started with a first iteration and see how it works :+1:

**Comment by JavierSegoviaCordoba** [comment]
Yeah, I am using multiple machines, if the other issue is closed before, this can be easier to be solved.

**Comment by JavierSegoviaCordoba** [comment]
In Kotlin Multiplatform there is no need to use multiple machines since 1.6.0. So verification with only one machine would be great.

Not sure if there are use cases where it is necessary to use multiple machines or if this requirement only existed with KMP

**Comment by martinbonnin** [comment]
Should we open an issue on the [Sonatype Jira](https://issues.sonatype.org/secure/Dashboard.jspa) about missing artifacts? 

Having to verify artifacts after a successful feels like a workaround. Of course having multiple ways to address a problem is always better but it'd be nice if the root cause of this problem could be investigated.
